gradle-plugin/doc/src/main/asciidoc/inc/scenarios/_groovy.adoc

```diff
@@ -0,0 +1,185 @@
+
+[[groovy-scenario-image-entrypoint]]
+==== Entrypoint
+You can specify `ENTRYPOINT` for your image using `entryPoint` field in `image` > `build`. See <<misc-startup, Startup Arguments>> for more details.
+
+[[groovy-scenario-image-copy-files]]
+==== Copying Files/Directories into Image
+If you want to copy some files/directories into your image. You can make use of {plugin} <<config-image-build-assembly, Assembly Configuration>>. You would need to provide `assembly` element in `image` > `build`.
+
+Here is an example of copying a single jar file into image:
+
+.Example for copying a file to image using Groovy DSL configuration:
+[source,groovy,subs="attributes+"]
+----
+kubernetes {
+    images {
+        image {
+            name = "${project.group}/${project.name}:${project.version}"
+            build {
+                from = "quay.io/jkube/jkube-java-binary-s2i:0.0.9"
+                assembly {
+                    targetDir = "/deployments"
+                    layers = [{
+                        id = "custom-assembly-for-copying-file"
+                        files = [{
+                            source = file("build/libs/${project.name}-${project.version}-all.jar")
+                            outputDirectory = "."
+                        }]
+                    }]
+                }
+            }
+        }
+    }
+}
+----
+
+In order to copy directories, you would be using `fileSets` configration element instead of `files`. Here is an example of copying directories:
+
+.Example for copying a directory to image using Groovy DSL configuration:
+[source,groovy,subs="attributes+"]
+----
+kubernetes {
+    images {
+        image {
+            name = "${project.group}/${project.name}:${project.version}"
+            build {
+                from = "quay.io/jkube/jkube-java-binary-s2i:0.0.9"
+                assembly {
+                    targetDir = "/deployments"
+                    layers = [{
+                        id = "custom-assembly-for-copying-directory"
+                        fileSets = [{
+                            directory = file("${project.rootDir}/build/dependencies")
+                       }]
+                    }]
+                }
+            }
+        }
+    }
+}
+----
+
+[[groovy-scenario-image-expose-container-ports]]
+==== Exposing Container Ports
+You can provide the ports you want to expose using `ports` field in `image` > `build`. Here is an example:
+
+.Example for exposing ports in image using Groovy DSL configuration:
+[source,groovy,subs="attributes+"]
+----
+kubernetes {
+    images {
+        image {
+            name = "${project.group}/${project.name}:${project.version}"
+            build {
+                // ...
+                ports = ['8080']
+            }
+        }
+    }
+}
+----
+
+
+You can read more about supported fields in `image` configuration element in <<config-image, Image Configuration>> section.
```

Comment:
The idea was to provide a thorough (numbered) image(s) configuration, and then describe what each point does.
For each point the description should link to the k8sBuild task Configuration section

Reply:
Maybe for the assembly part, an additional example is interesting, but individualized examples for every single configuration point is too much.